agent/src/softwarecontaineragent.h

```diff
@@ -47,6 +48,104 @@ namespace softwarecontainer {
 
 static constexpr ContainerID INVALID_CONTAINER_ID = -1;
 
+class SoftwareContainerAgentError : public SoftwareContainerError
+{
+public:
+    SoftwareContainerAgentError():
+        m_message("SoftwareContainer error")
+    {
+    }
+
+    SoftwareContainerAgentError(const std::string &message):
+        m_message(message)
+    {
+    }
+
+    virtual const char *what() const throw()
+    {
+        return m_message.c_str();
+    }
+
+protected:
+    std::string m_message;
+};
+
+class PreloadError : public SoftwareContainerAgentError
+{
+public:
+    PreloadError():
+        SoftwareContainerAgentError("Software Container error - Failed to preload container")
+    {
+    }
+
+    PreloadError(const std::string &message):
+        SoftwareContainerAgentError(message)
+    {
+    }
+
+    virtual const char *what() const throw()
+    {
+        return SoftwareContainerAgentError::m_message.c_str();
```

Comment:
Since the parent is initialized with the message, shouldn't it be enough to just let the method be in the parent and not implement it again here?